opengrok-indexer/src/main/java/org/opengrok/indexer/web/messages/Message.java

```diff
@@ -157,6 +191,28 @@ public int compareTo(final Message o) {
         return tags.size() - o.tags.size();
     }
 
+    private static class MessageLevelDeserializer extends StdDeserializer<MessageLevel> {
+        private static final long serialVersionUID = 928540953227342817L;
+
+        MessageLevelDeserializer() {
+            this(null);
+        }
+
+        MessageLevelDeserializer(Class<?> vc) {
+            super(vc);
+        }
+
+        @Override
+        public MessageLevel deserialize(final JsonParser parser, final DeserializationContext context)
+                throws IOException {
+            try {
+                return MessageLevel.stringToMessageLevel(context.readValue(parser, String.class));
+            } catch (DateTimeParseException e) {
+                throw new IOException(e);
+            }
+        }
+    }
+
```

Comment:
I just wonder, serializer is not needed?

Reply:
nope, it works out of the box.